src/main/webapp/WEB-INF/view/profiles.jsp

```diff
@@ -45,10 +45,10 @@ User user = (User) request.getAttribute("user");
 
     <h1>Profiles</h1>
 
-<% List<User> list = UserStore.getInstance().getUsersList();
+<% List<User> list = userStore.getUsersList();
     for (int i = 0; i < list.size(); i++) {
       String currUser = list.get(i).getName(); %>
-      <li>User name: <%= currUser %> </li>
+      <li>User name: <a href= \<%= "profile/" + currUser %> > <%= currUser %></a> </li>
 <% } %>
 
```

Comment:
You changed UserStore.getInstance().getUsersList(); to List<User> list = userStore.getUsersList(); which failed to compile when i had that originally, so unless you changed the original value to static this might crash in tests, have you tried running them?

Reply:
Yeah I ran tests on these. What happened with your code is you were requesting "user" instead of "users". if you look at line 22 of profiles.jsp, you'll see I changed the request to be for "users" so it now properly receives that data. Tl;dr : It works haha.